engines/suppliers/app/models/suppliers/order.rb

```diff
@@ -1,10 +1,36 @@
-module Supplier
+module Suppliers
   class Order < ActiveRecord::Base
     self.table_name = :orders
 
     has_many :order_lines, class_name: 'Suppliers::OrderLine'.freeze
 
-    belongs_to :user, class_name: 'Suppliers::User'.freeze
-    belongs_to :group, class_name: 'Suppliers::Group'.freeze
+    belongs_to :from_user, class_name: 'Suppliers::User'.freeze
+    belongs_to :from_group, class_name: 'Suppliers::Group'.freeze
+    belongs_to :to_group, class_name: 'Suppliers::Group'.freeze
+    belongs_to :to_producer, class_name: 'Suppliers::Producer'.freeze
+
+    validate :actor_presence
+    validate :target_presence
+    validates :confirm_before, presence: true
+
+    private
+
+    # Validates `from_user_id` xor `from_group_id` presence
+    #
+    def actor_presence
+      return if from_user_id.blank? ^ from_group_id.blank?
```

Comment:
it's an [exclusive OR](https://en.wikipedia.org/wiki/Exclusive_or): `a ^ b` means that `a` or `b` must exists but not both, just one of them.